rustls/src/client/tls12.rs

```diff
@@ -40,29 +40,45 @@ mod server_hello {
     use super::*;
     use crate::client::hs::{ClientHelloInput, ClientSessionValue};
     use crate::msgs::handshake::ServerHelloPayload;
+    use crate::sealed::Sealed;
 
-    pub(in crate::client) struct CompleteServerHelloHandling {
-        pub(in crate::client) randoms: ConnectionRandoms,
-        pub(in crate::client) transcript: HandshakeHash,
-        pub(in crate::client) input: ClientHelloInput,
-    }
+    pub(crate) static TLS12_HANDLER: &dyn Tls12Handler = &Handler;
 
-    impl CompleteServerHelloHandling {
-        pub(in crate::client) fn handle_server_hello(
-            mut self,
+    pub(crate) trait Tls12Handler: Sealed + Send + Sync + core::fmt::Debug {
```

Comment:
Trait definition below type implementing it? Maybe move `fmt::Debug` before `Send + Sync` (because more notable) and import `fmt` at the top of the module?

In terms of the commit history, would suggest splitting this first into a free function, to deal with all the `self.` removals, and then introducing the `Handler`.